### PR TITLE
MS-180 Cover code with unit tests

### PR DESCRIPTION
**Pull request by SVilgelm**
Modify main.py script to allow using a custom configuration file and
enabling debug mode.
Implement the unit tests.

### CODE REVIEW

File: health/main.py [hunk 1]
```diff
@@ -121,17 +119,19 @@ def job():
...
+    args = parser.parse_args(argv)
+    logging.basicConfig(format=LOGGING_FORMAT, level=args.debug)
+
     global CONF
     try:
-        with open(CONF_PATH) as f:
-            CONF = json.load(f)
-            jsonschema.validate(CONF, CONF_SCHEMA)
-
-    except (OSError, IOError):
```
Comment (boris-42) [inline health/main.py]: I don't like this change, without it debugging is hard. If we would like to refactor this we can change print() to logs

File: health/main.py [hunk 2]
```diff
@@ -121,17 +119,19 @@ def job():
                      % (time.time() - started_at))
 
 
-def main():
+def main(argv=sys.argv[1:]):
```
Comment (boris-42) [inline health/main.py]: main() function should not accept any arguments, otherwise you can't use it as entry point

File: tests/unit/health/mapping/test_es.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+# from health.mapping import es
+
+from tests.unit import test
+
+
+class InitElasticTestCase(test.TestCase):
```
Comment (boris-42) [inline tests/unit/health/mapping/test_es.py]: This should go like a separated commit (because it's change of unit test that is not related to changes in main)

File: tests/unit/health/mapping/test_es.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    def test(self):
```
Comment (boris-42) [inline tests/unit/health/mapping/test_es.py]: You don't need this method..  just add "pass" to class..

File: health/main.py (hunk 2 above)
Comment (SVilgelm) [inline health/main.py]: That's why the `argv` variable  has a default value.

File: tests/unit/health/mapping/test_es.py (hunk 3 above)
Comment (SVilgelm) [inline tests/unit/health/mapping/test_es.py]: It's is not, but i need thus changes to cover `main` function with the unit tests. Let me finish the work tomorrow.

File: tests/unit/health/mapping/test_es.py (hunk 4 above)
Comment (SVilgelm) [inline tests/unit/health/mapping/test_es.py]: I know, thanks :) It's just a draft.

File: health/main.py (hunk 2 above)
Comment (boris-42) [inline health/main.py]: 1) Please do not use any argument that is important. 
2) Do not use mutable default values for arguments. This is python anitpattern http://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html 

So just remove it and put as first string 

  argv = sys.argv[1:]

That will also simplify testing this method (because mocking will become way simpler)

File: tests/unit/health/mapping/test_es.py (hunk 3 above)
Comment (boris-42) [inline tests/unit/health/mapping/test_es.py]: take a look at my comment above, about main function=)

File: health/main.py (hunk 1 above)
Comment (SVilgelm) [inline health/main.py]: Hmm, `argparse` library does these checks instead of us and shows an error. But i can revert it back.

Comment (SVilgelm) [inline health/main.py]:
```
venv runtests: commands[0] | health -c etc/config.json1
usage: health [-h] [-c CONFIG_PATH] [-d]
health: error: argument -c/--config-path: can't open 'etc/config.json1': [Errno 2] No such file or directory: 'etc/config.json1'
```

**Comment by boris-42** [comment]
code changed a lot, tests were merged in another pull request
